ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/common/FailureXAResource.java

```diff
@@ -9,157 +9,131 @@
-    {
+
+    public FailureXAResource(FailLocation loc, FailType type) {
         _locale = loc;
         _type = type;
     }
-    
-    public void commit(Xid id, boolean onePhase) throws XAException
-    {
-	if (_locale == FailLocation.commit)
-	{
-	    if (_type == FailType.normal)
-	        throw new XAException(XAException.XA_HEURMIX);
-	    
-	    if (_type == FailType.heurcom)
-	        throw new XAException(XAException.XA_HEURCOM);
-	    
-	    if (_type == FailType.rollback)
-                throw new XAException(XAException.XA_HEURRB);
-	    
-	    if (_type == FailType.nota)
-	        throw new XAException(XAException.XAER_NOTA);
-	    
-	    if (_type == FailType.inval)
-                throw new XAException(XAException.XAER_INVAL);
-	    
-	    if (_type == FailType.proto)
-                throw new XAException(XAException.XAER_PROTO);
-	    
-	    if (_type == FailType.rmfail)
-                throw new XAException(XAException.XAER_RMFAIL);
-	    
-        if (_type == FailType.XA_RBINTEGRITY)
-            throw new XAException(XAException.XA_RBINTEGRITY);
-
-	    throw new XAException(XAException.XA_RBTIMEOUT);
-	}
-    // successful
-    this._committed = true;
+
+    public void commit(Xid id, boolean onePhase) throws XAException {
+        if (_locale == FailLocation.commit) {
+            switch (_type) {
+                case normal:
+                    throw new XAException(XAException.XA_HEURMIX);
+                case heurcom:
+                    throw new XAException(XAException.XA_HEURCOM);
+                case rollback:
+                    throw new XAException(XAException.XA_HEURRB);
+                case nota:
+                    throw new XAException(XAException.XAER_NOTA);
+                case inval:
+                    throw new XAException(XAException.XAER_INVAL);
+                case proto:
+                    throw new XAException(XAException.XAER_PROTO);
+                case rmfail:
+                    throw new XAException(XAException.XAER_RMFAIL);
+                case XA_RBINTEGRITY:
+                    throw new XAException(XAException.XA_RBINTEGRITY);
+                default:
+                    throw new XAException(XAException.XA_RBTIMEOUT);
+            }
+        }
+        // successful
+        this._committed = true;
     }
 
-    public void end(Xid xid, int flags) throws XAException
-    {
-        if (_locale == FailLocation.end)
-        {
-            if (_type == FailType.normal)
-                throw new XAException(XAException.XA_HEURRB);
-            
-            if (_type == FailType.timeout)
-                throw new XAException(XAException.XA_RBTIMEOUT);
-            
-            if (_type == FailType.XA_RBCOMMFAIL)
-                throw new XAException(XAException.XA_RBCOMMFAIL);
+    public void end(Xid xid, int flags) throws XAException {
+        if (_locale == FailLocation.end) {
+            switch (_type) {
```

Comment:
Noting here 2/3 changes from if -> switch

Reply:
I rolled back this change